xla/service/gpu/fusions/mlir/mlir_fusion_emitter.cc

```diff
@@ -590,8 +593,22 @@ void AddLoweringPasses(mlir::OpPassManager& pm,
   pm.addPass(mlir::createLoopInvariantCodeMotionPass());
   pm.addPass(mlir::createSymbolDCEPass());
   pm.addPass(mlir::createCSEPass());
+
+  auto ptx_version =
+      se::GetAsmCompilerVersion(debug_options.xla_gpu_cuda_data_dir());
   pm.addPass(CreateExpandFloatOpsPass(
-      !device.cuda_compute_capability().IsAtLeastAmpere()));
+      /*pre_ampere=*/!device.cuda_compute_capability().IsAtLeastAmpere(),
+      /*has_f8_cvt=*/!ptx_version.ok()
```

Comment:
Could you please extract this logic into a function with several if conditions? I think that would make it easier to read.

Reply:
Yeah, I think that's probably better. We're running into a bunch of RoCM build failures, having a separate pass might make that more manageable without littering ifdefs all over the place.

Reply:
Extracting into a function with `if`s is clear to me, but adding a separate pass is not - could you elaborate?

Reply:
I was imagining an Nvidia-specific pass that runs before this one, replacing all the conversions that we have efficient intrinsics for, so this pass only cleans up the rest afterwards. Then we can just add that one to the pipeline if we're compiling for Nvidia. I'm worried if we add all these switches in this pass, it will get a bit difficult to follow.

WDYT?

Reply:
Makes sense, will update this PR soon.

Reply:
Extracted the NVidia-specific logic into a separate (optional) pass. PTAL

Reply:
Awesome, thank you so much! Looks great.